web/entrypoint.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env sh
+set -e
+
+gen_self_signed() {
+  echo "[WRN] No certificates found, generating self signed cert with key"
+  openssl req -x509 -nodes -days 1780 -newkey rsa:4096 \
+    -keyout /etc/ssl/certs/tea.key \
+    -out /etc/ssl/certs/tea_bundle.crt \
+    -subj "/C=DE/ST=Berlin/L=Germany/O=TeaSpeak/OU=TeaWeb/CN=localhost/emailAddress=noreply@teaspeak.de"
+}
+
+gen_diffie_hellman() {
+  echo "[INF] No Diffie-Hellman pem found, generating new with 2048 byte"
+  openssl dhparam -out /etc/ssl/certs/dhparam.pem 2048
+}
+
+if [ "$1" = "nginx" ]; then
+  if [ ! -f /etc/ssl/certs/tea.key ] && [ ! -f /etc/ssl/certs/tea_bundle.crt ]; then
+    gen_self_signed
+  elif [ ! -f /etc/ssl/certs/tea.key ] || [ ! -f /etc/ssl/certs/tea_bundle.crt ]; then
+    echo "[ERR] Only found a key or crt-bundle file but both files are REQUIRED!"
+    exit 1
```

Comment:
Combined with the `--restart=unless-stopped` wouldn't it cause an infinity restart loop?
Is it may somehow possible to signal back, that the container isn't "healthy", so docker does not (re)starts it?

Reply:
Well, yes even though this can indeed happen the user should notice this rather quickly. (eg. not being able to connect to web client -> reads logs -> understands whats wrong)
Alternatively, I can only remove the option, as there is not [restart policy](https://docs.docker.com/engine/reference/run/#restart-policies---restart) that would stop restarting after x failed attempts.
IMHO this case is neglectable, since Nginx would not be able to boot up if I wouldn't have this check and force the container to stop/restart that way (as its the process keeping the container alive).

To answer your second question, yes there is a `HEALTHCHECK` instruction available for docker containers, but it does not influence the restart policy.
Additionally, since Nginx offers no way per default to check if it's healthy I would have to rely on a custom `/healtcheck` endpoint or the web client offering something to that extend.
Admittedly I would love to add a `HEALTHCHECK` to both the server as well as the WebClient image, yet neither the server nor the client offers a good endpoint to query for such a metric.

Reply:
Well an auto restart, especially for the server would be really welcomed, at least most users think so ;)

My approach would be to test if the process is still alive.
This might be extended by a simple "query" check alive check, which should not be so hard to script. I could provide such scripts later, but currently busy.

Reply:
No explicit health check needed